internal/versions/versions.go

```diff
@@ -338,6 +358,83 @@ func (h *handler) validateVersions() error {
 	return nil
 }
 
+func existAlready(db *gorm.DB, openShiftReleaseImage common.OpenShiftReleaseImage) bool {
+	var count int64
+	db.Model(&common.OpenShiftReleaseImage{}).
+		Where("openshift_version = ? AND channel = ? AND cpu_architecture = ?", openShiftReleaseImage.OpenshiftVersion, openShiftReleaseImage.Channel, openShiftReleaseImage.CpuArchitecture).
+		Count(&count)
+
+	return count > 0
+}
+
+func syncReleaseImages(db *gorm.DB, releaseSources models.ReleaseSources) error {
+	for _, releaseSource := range releaseSources {
+		openshiftVersion := *releaseSource.OpenshiftVersion
+		for _, upgradeChannel := range releaseSource.UpgradeChannels {
+			cpuArchitecture := *upgradeChannel.CPUArchitecture
+			for _, channel := range upgradeChannel.Channels {
+				// OCP releases API needs amd64, arm64 instead of x86_64, aarch64 respectively
+				if cpuArchitecture == "x86_64" {
+					cpuArchitecture = "amd64"
+				} else if cpuArchitecture == "aarch64" {
+					cpuArchitecture = "arm64"
+				}
+				endpoint := fmt.Sprintf(VersionsAPITemplate, channel, openshiftVersion, cpuArchitecture)
+
+				resp, err := http.Get(endpoint)
+				if err != nil {
+					return errors.Errorf("an error occured while making http request to %s: %s", endpoint, err)
+				}
+
+				// switch back
+				if cpuArchitecture == "amd64" {
+					cpuArchitecture = "x86_64"
+				} else if cpuArchitecture == "arm64" {
+					cpuArchitecture = "aarch64"
+				}
+
+				var graph Graph
+				err = json.NewDecoder(resp.Body).Decode(&graph)
+				if err != nil {
+					return errors.Errorf("an error occured while decoding the response to a request made to %s: %s", endpoint, err)
+				}
+
+				for _, node := range graph.Nodes {
+					if strings.HasPrefix(node.Version, openshiftVersion) {
+						// Got a match for OpenShiftReleaseImage
+						releaseUrl := fmt.Sprintf(OcpImageUrlTemplate, node.Version, cpuArchitecture)
+						openShiftReleaseImage := common.OpenShiftReleaseImage{
+							OpenshiftVersion: node.Version,
+							Channel:          channel,
+							CpuArchitecture:  cpuArchitecture,
+							Url:              releaseUrl,
+						}
+
+						if existAlready(db, openShiftReleaseImage) {
+							continue
+						}
+						db.Create(&openShiftReleaseImage)
```

Comment:
Could some image cease to exist? Thinking early access here, but there might be other cases. How are we going to handle those?

Reply:
That is a good point. I am not sure it can happen, but either way flushing the table each run is a good idea. WDYT?

Reply:
I think maybe flushing, given we can safely lock the table for a small amount of time... but we might risk to be without data if things go wrong.
Another option would be having an internal field transaction ID or something, update all records then delete the ones that don't have such transaction ID.

Reply:
Not sure I understand, we can just update then if it is already exists no?

Reply:
for the second option, yes, we can update if it already exists, and then delete all the entries with different transaction/version ID, i.e.:
```
version | arch | versionID |
--------+------+-----------|
foobar  | x86  | 0af...    |
foobar  | arm  | 0af...    |
```

next update we only get `foobar, arm`:
```
version | arch | versionID |
--------+------+-----------|
foobar  | x86  | 0af...    |
foobar  | arm | 123...     |
```
then delete all records with versionID != 123

another way could be what you suggested about flushing, but my concern is the risk of locking or leaving empty. However the versionID solution risks of reporting stale data.